OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -68,6 +71,13 @@ public void Add(Player viewer, FrozenActor fa)
 			partitionedFrozenActors[viewer].Add(fa, FrozenActorBounds(fa));
 		}
 
+		public void Add(IEffect effect, WPos position, Rectangle bounds)
```

Comment:
Several of my earlier comments raise issues about passing in `Rectangles` that have a bogus origin.
I just noticed here that you aren't even using the origin here or in `Update`, so `Rectangle` is the wrong thing to use.

Please replace these with two overloads that take a `Sprite` or a `Size`, solving the need to define dummy positions that can cause confusion.